### PR TITLE
version: update latest from dev to 23.2

### PR DESCRIPTION
**Pull request by Deflaimun**
PRs linked to 23.2
https://github.com/redpanda-data/documentation/pull/1525/files
https://github.com/redpanda-data/documentation/pull/1685/files
https://github.com/redpanda-data/documentation/pull/1671/files
https://github.com/redpanda-data/documentation/pull/1658/files
https://github.com/redpanda-data/documentation/pull/1633/files
https://github.com/redpanda-data/documentation/pull/1606/files
https://github.com/redpanda-data/documentation/pull/1596/files
https://github.com/redpanda-data/documentation/pull/1592/files
https://github.com/redpanda-data/documentation/pull/1587/files
https://github.com/redpanda-data/documentation/pull/1564/files
https://github.com/redpanda-data/documentation/pull/1542/files

Files that are present in 23.2 PRs
```
versioned_docs/version-23.2/manage/monitoring.mdx
versioned_docs/version-23.2/reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx
versioned_docs/version-23.2/get-started/architecture.mdx
versioned_docs/version-23.2/manage/kubernetes/tiered-storage.mdx
versioned_docs/version-23.2/manage/shared/tiered-storage/_archived-metadata.mdx
versioned_docs/version-23.2/manage/tiered-storage.mdx
versioned_docs/version-23.2/deploy/deployment-option/self-hosted/manual/production/production-deployment.mdx
versioned_docs/version-23.2/manage/cluster-maintenance/cluster-balancing.mdx
versioned_docs/version-23.2/reference/cluster-properties.mdx
versioned_docs/version-23.2/manage/shared/tiered-storage/_space-usage.mdx
versioned_docs/version-23.2/reference/tunable-properties.mdx
versioned_docs/version-23.2/manage/shared/tiered-storage/_caching.mdx
versioned_docs/version-23.2/manage/shared/tiered-storage/_remote-read.mdx
versioned_docs/version-23.2/manage/cluster-maintenance/disk-utilization.mdx
versioned_docs/version-23.2/manage/shared/tiered-storage/_set-retention-limits.mdx
versioned_docs/version-23.2/manage/shared/tiered-storage/_upload-controller.mdx
versioned_docs/version-23.2/get-started/licenses.mdx
versioned_docs/version-23.2/manage/schema-id-validation.mdx
versioned_sidebars/version-23.2-sidebars.json
versioned_docs/version-23.2/develop/consume-data/follower-fetching.mdx
versioned_docs/version-23.2/develop/consume-data/index.mdx
versioned_docs/version-23.2/develop/kafka-clients.mdx
versioned_docs/version-23.2/manage/console/topic-documentation.mdx
```

### CODE REVIEW

File: versioned_docs/version-23.2/manage/kubernetes/tiered-storage.mdx [hunk 1]
```diff
@@ -387,7 +389,7 @@ For additional properties, see [Tiered Storage configuration properties](#tiered
 
 <Reconciliation frontmatter={frontMatter}/>
 
-### Upload backlog controller
+### Upload controller
```
Comment (micheleRP) [inline versioned_docs/version-23.2/manage/kubernetes/tiered-storage.mdx:392]: In 23.2, this heading should be Upload backlog controller

Comment (Deflaimun) [reply versioned_docs/version-23.2/manage/kubernetes/tiered-storage.mdx]: Fixed

**Review by shorgi — COMMENTED** [review]
cluster and tunable properties LGTM, they're consistent with what's currently in dev

**Review by micheleRP — APPROVED** [review]
lgtm

**Review by Feediver1 — APPROVED** [review]
lgtm

**Pull request merged**